sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContext.java

```diff
@@ -96,6 +100,36 @@ private ParallelDocumentQueryExecutionContext(
         }
     }
 
+    public static <T extends Resource> Flux<IDocumentQueryExecutionComponent<T>> createReadManyQueryAsync(
+        IDocumentQueryClient queryClient,
+        String collectionResourceId, SqlQuerySpec sqlQuery,
+        Map<PartitionKeyRange, String> rangeQueryMap,
+        FeedOptions feedOptions, String collectionRid, String collectionLink, UUID activityId, Class<T> klass,
+        ResourceType resourceTypeEnum) {
+        
+        List<PartitionKeyRange> ranges = new ArrayList<>();
```

Comment:
same about this one too. shouldn't this be thread-safe?

Reply:
It will be modified only inside this flatmap and is called only once.